lib/spack/spack/environment.py

```diff
@@ -0,0 +1,200 @@
+import os
+import os.path
+import collections
+
+
+class AttributeHolder(object):
+    """
+    Policy that permits to store any kind of attribute on self. The attributes must be passed as key/value pairs
+    during the initialization of the instance.
+    """
+    def __init__(self, **kwargs):
```

Comment:
So maybe keep it and do like this?

class NameValueModifier(object):
    def __init__(self, name, value, **kwargs):
        self.name = name
        self.value = value
        self.format_args = { 'name' : name, 'value' : value, 'path' : value }
        self.format_args.update(kwargs)

class SetEnv(NameValueModifier):
    def execute(self):
        os.environ[self.name] = str(self.value)